advanced/Templates/gravity.db.sql

```diff
@@ -81,48 +94,52 @@ CREATE TABLE info
 
 INSERT INTO info VALUES("version","1");
 
-CREATE VIEW vw_gravity AS SELECT a.domain
-    FROM gravity a
-    WHERE a.domain NOT IN (SELECT domain from vw_whitelist);
+CREATE VIEW vw_gravity AS SELECT domain
+    FROM gravity
+    WHERE domain NOT IN (SELECT domain from vw_whitelist);
 
-CREATE VIEW vw_whitelist AS SELECT a.domain
-    FROM whitelist a
-    INNER JOIN domain_groups b ON b.id = a.group_id
-    WHERE a.enabled = 1 AND b.enabled = 1
-    ORDER BY a.id;
+CREATE VIEW vw_whitelist AS SELECT domain
+    FROM whitelist
+    LEFT JOIN whitelist_by_group ON whitelist_by_group.whitelist_id = whitelist.id
+    LEFT JOIN domain_groups ON domain_groups.id = whitelist_by_group.group_id
+    WHERE whitelist.enabled = 1 AND domain_groups.enabled IS NULL OR domain_groups.enabled == 1
```

Comment:
`AND` has precedence over `OR` (`a AND b OR c` is `(a AND b) OR c`), and `domain_groups.enabled` doesn't make much sense at first. This may be better written as:
```SQL
WHERE whitelist.enabled = 1 AND (whitelist_by_group.group_id IS NULL OR domain_groups.enabled = 1)
```

This applies to the other similar expressions as well. Also note that standard SQL uses `=` instead of `==`, even though SQLite allows the latter.

Reply:
agreed.

Reply:
`WHERE` clause should be:

```suggestion
    WHERE whitelist.enabled = 1 AND domain_groups.enabled != 0
```

Reply:
Note we found that the above has an issue because you cannot compare a value with NULL: https://stackoverflow.com/questions/5658457/not-equal-operator-on-null

Reply:
This does not work as `(NULL != 0)` is always `false` in SQL.